Userland/Services/SpiceAgent/Message.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <AK/Vector.h>
+
+namespace SpiceAgent {
+
+static constexpr u32 AGENT_PROTOCOL = 1;
+
+// Used to communicate what the client/or server is capable of
+// Not a lot of documentation is available for all of these, but the headers contain some information
+// https://gitlab.freedesktop.org/spice/spice-protocol/-/blob/master/spice/vd_agent.h
+enum class Capability : u32 {
+    MouseState = 0,
+    MonitorsConfig,
+    Reply,
+    Clipboard,
+    DisplayConfig,
+    ClipboardByDemand,
+    ClipboardSelection,
+    SparseMonitorsConfig,
+    GuestLineEndLF,
+    GuestLineEndCRLF,
+    MaxClipboard,
+    AudioVolumeSync,
+    MonitorsConfigPosition,
+    FileTransferDisabled,
+    FileTransferDetailedErrors,
+    GraphicsCardInfo,
+    ClipboardNoReleaseOnRegrab,
+    ClipboardGrabSerial
+};
+
+class Message {
+public:
+    // The spice protocol headers contain a bit of documentation about these, but nothing major:
+    // https://gitlab.freedesktop.org/spice/spice-protocol/-/blob/master/spice/vd_agent.h
+    enum class Type : u32 {
+        MouseState = 1,
+        MonitorsConfig,
+        Reply,
+        Clipboard,
+        DisplayConfig,
+        AnnounceCapabilities,
+        ClipboardGrab,
+        ClipboardRequest,
+        ClipboardRelease,
+        FileTransferStart,
+        FileTransferStatus,
+        FileTransferData,
+        Disconnected,
+        MaxClipboard,
+        VolumeSync,
+        GraphicsDeviceInfo
+    };
+
+    Message(Type type)
+        : m_type(type)
+    {
+    }
+
+    Type type() { return m_type; }
+
+    // Used for sending messages to the spice server
+    virtual ErrorOr<void> write_to_stream(AK::Stream& stream);
+
+    // Used for debugging
+    virtual ErrorOr<String> debug_description();
+
+    virtual ~Message() = default;
+
+private:
+    Type m_type;
```

Comment:
This should be initialized to a known value (because this is a primitive type).

Reply:
What would be a good one for this? 
Should I add an `Unknown` message type?

Reply:
Not sure adding `Unknown` type is a good thing, unless we know how to handle that properly. A valid option is to use the first option in the enum (as a placeholder here), but sometimes it will be awkward and bug-prone too.